Hmac.js

```diff
@@ -48,12 +54,45 @@ export class Hmac {
    *
    * @param {object} options - The options to use.
    * @param {Uint8Array} options.data - The data to sign as a Uint8Array.
+   * @param {boolean} options.useCache - Enable the use of a cache.
    *
    * @returns {Promise<string>} The base64url-encoded signature.
    */
-  async sign({data}) {
+  async sign({data, useCache = true}) {
+    const cacheKey = `sign-${base64url.encode(data)}`;
+    if(useCache) {
+      const signature = this.cache.get(cacheKey);
+      if(signature) {
+        return signature;
+      }
+    }
+
     const {id: keyId, kmsClient, capability, invocationSigner} = this;
-    return kmsClient.sign({keyId, data, capability, invocationSigner});
+    const promise = kmsClient.sign({keyId, data, capability, invocationSigner});
+
+    if(useCache) {
+      // 1. Set promise in cache
+      this.cache.set(cacheKey, promise);
+
+      // 2. Clear existing timer
+      if(this._pruneCacheTimer) {
+        clearTimeout(this._pruneCacheTimer);
+      }
```

Comment:
This doesn't seem right -- this will continually push the timer out and could even do so indefinitely (in a busy system) such that pruning never happens. I think all we want to do here is set the timer if it isn't currently set.